service/common/src/main/java/org/apache/polaris/service/catalog/io/FileIOFactory.java

```diff
@@ -18,10 +18,50 @@
  */
 package org.apache.polaris.service.catalog.io;
 
+import jakarta.annotation.Nonnull;
+import jakarta.enterprise.context.ApplicationScoped;
 import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.io.FileIO;
+import org.apache.polaris.core.context.RealmId;
+import org.apache.polaris.core.persistence.PolarisResolvedPathWrapper;
+import org.apache.polaris.core.storage.PolarisStorageActions;
 
-/** Interface for providing a way to construct FileIO objects, such as for reading/writing S3. */
+/**
+ * Interface for providing a way to construct FileIO objects, such as for reading/writing S3.
+ *
+ * <p>Implementations are available via CDI as {@link ApplicationScoped @ApplicationScoped} beans.
```

Comment:
nit: why mention `@ApplicationScoped` on the interface that does not carry this annotation? IMHO, this is not an API concern, but an integration/build concern.

Reply:
Suggested by @snazy in this comment: https://github.com/apache/polaris/pull/724#discussion_r1924753312

Reply:
I'm fine to keep this comment :)